massa-execution-worker/src/speculative_async_pool.rs

```diff
@@ -221,9 +221,14 @@ impl SpeculativeAsyncPool {
         }
 
         // Query eliminated messages
-        let eliminated_msg =
+        let mut eliminated_msg =
             self.fetch_msgs(eliminated_infos.iter().map(|(id, _)| id).collect(), true);
 
+        eliminated_msg.extend(eliminated_new_messages.iter().filter_map(|(k, v)| match v {
```

Comment:
I don't understand this. Aren't the `eliminated_new_messages` already taken into account in `eliminated_msg` at line 187 ?

Reply:
The problem is that if the message is emitted and eliminated in the same slot, it will not be in the changes above and not in the eliminated_new_messages returned.